flake.nix

```diff
@@ -255,13 +255,30 @@
       lib ? pkgs.lib,
       check ? true,
       extraSpecialArgs ? {},
-    }:
-      modulesWithInputs {
+    }: let
+      output = modulesWithInputs {
         inherit pkgs lib check extraSpecialArgs;
         configuration = {...}: {
           imports = modules;
         };
       };
+    in
+      output
+      // {
+        neovim =
+          output.neovim
+          // {
+            extendConfiguration = args:
+              (neovimConfiguration {
+                modules = modules ++ (args.modules or []);
+                pkgs = args.pkgs or pkgs;
+                lib = args.lib or lib;
+                check = args.check or check;
+                extraSpecialArgs = extraSpecialArgs // (args.extraSpecialArgs or {});
+              })
+              .neovim;
```

Comment:
```suggestion
```

From my understanding, the result of the function does not require being `.neovim`. I would rather the output of `extendConfiguration` be identical to `neovimConfiguration` (as the module exports some useful information).

Reply:
Rebasing this PR on #30 should provide the perfect solution. You can make `extendConfiguration` a `passthru` [here](https://github.com/jordanisaacs/neovim-flake/pull/30/files#diff-8602cc26f03b5931a12522c6caa96527548a205f9a1c76745fd2a1b4567ff1d8R138). And pass the `_useBaseModules` = false to the neovimConfiguration function.

Reply:
If that is out of your capacity I can take this PR over but I do not want to step on your work as this is a great solution.

Reply:
Please, feel free to take it over; 10mo baby here is requiring additional time this week, and I won't have time to implement for a couple of days. I'm happy to have contributed at least the idea, but of course you as creator has the best big picture view to make it right. Thanks